core/trino-parser/src/main/antlr4/io/trino/sql/parser/SqlBase.g4

```diff
@@ -406,6 +426,9 @@ primaryExpression
     | POSITION '(' valueExpression IN valueExpression ')'                                 #position
     | '(' expression (',' expression)+ ')'                                                #rowConstructor
     | ROW '(' expression (',' expression)* ')'                                            #rowConstructor
+    | name=LISTAGG '(' setQuantifier? expression (',' string)?
+        (ON OVERFLOW listAggOverflowBehavior)? ')'
+        (WITHIN GROUP '(' ORDER BY sortItem (',' sortItem)* ')')? listaggOver?            #listagg
```

Comment:
The `within group specification` part of the synopsis of the `LISTAGG` function was missing from #4835 so I just improvised based on the content which I've found on the Internet about the specification of the function (I don't have at hand the SQL:2016 standard document).

According to https://modern-sql.com/feature/listagg

> The over clause must not contain an order by clause because the mandatory within group clause must contain an order by clause anyway.

See SQL:2016-2: §6.10, Syntax Rule 12

Can you please verify whether the SQL standard does indeed allow this exception for `LISTAGG` ?

Reply:
That rule says:

> ) If the window ordering clause or the window framing clause of the window structure descriptor that describes the <window name or specification> is present, then no `<aggregate function>` simply contained in `<window function>` shall specify DISTINCT or `<ordered set function>`.

So, technically speaking, if the window has an ordering or framing, `listagg` cannot be used at all (it's an "ordered set" function). It's not clear why it places such constraint -- it could be do to semantic issues (which I need to investigate further) or because it would make an implementation hard or potentially very inefficient.

Reply:
Is within group actually mandatory? 
I have it in my current version optional.

Reply:
As discussed over Slack with @martint in the initial phase, the `LISTAGG` function will not support window framing clause.